connmgr/seed.go

```diff
@@ -60,14 +60,16 @@ func SeedFromDNS(chainParams *chaincfg.Params, reqServices wire.ServiceFlag,
 			// if this errors then we have *real* problems
 			intPort, _ := strconv.Atoi(chainParams.DefaultPort)
 			for i, peer := range seedpeers {
-				addresses[i] = new(wire.NetAddress)
-				addresses[i].SetAddress(peer, uint16(intPort))
-				// bitcoind seeds with addresses from
-				// a time randomly selected between 3
-				// and 7 days ago.
-				addresses[i].Timestamp = time.Now().Add(-1 *
-					time.Second * time.Duration(secondsIn3Days+
-					randSource.Int31n(secondsIn4Days)))
+				addresses[i] = &wire.NetAddress{
+					// bitcoind seeds with addresses from
+					// a time randomly selected between 3
+					// and 7 days ago.
+					Timestamp: time.Now().Add(-1 *
```

Comment:
You're correct.  It's already doing a heap alloc anyways, so it certainly wouldn't hurt to refactor it into a func.  If you do add the func, please make sure you maintain the test coverage by ensuring the new func is covered by the tests as well.

On a somewhat related note, one of the things we'd like to ultimately do is reduce allocs by using `[]wire.X` instead of `[]*wire.X` where appropriate.  I believe `[]wire.NetAddress` versus `[]*wire.NetAddress` is one such case.  The first one is a single, contiguous allocation which is better for cache locality and reduces the burden on the GC, and translates into less memory usage over all since there is less tracking overhead needed for the GC for a bunch of small allocations.

Most of the `[]*wire.X` stuff is legacy from when we were new to Go and didn't have as clear of an understanding of the way memory was handled by the GC and also an artifact of how the `range` statement works.